tools/docker-utils.js

```diff
@@ -0,0 +1,149 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const tempy = require('tempy');
+const execa = require('execa');
+const { getUnleashServerVersion } = require('./unleash-server-utils');
+
+const tempDirPath = tempy.directory();
+
+/** @type {string} */
+const dockerfileTemplate = fs.readFileSync(
+  path.resolve(__dirname, '..', 'Dockerfile'),
+  { encoding: 'utf8' },
+);
+
+const fromLineRegExp = /FROM node:.*?\n/g;
+
+/**
+ * Render a Dockerfile with the correct base image.
+ *
+ * @param nodeDockerVersion {string}
+ * @returns {string} the Dockerfile content with the correct base image
+ */
+function renderDockerfile({nodeDockerVersion}) {
+  return dockerfileTemplate.replace(fromLineRegExp, `FROM node:${nodeDockerVersion}`);
+}
```

Comment:
I initially messed around with a templating engine (such as `doT`), but figured I couldn't convert the Dockerfile to a template file without breaking the `docker-compose` setup, so I ended up just leaving the Dockerfile as is and regexp-ing around the problem

Reply:
For now it would probably also break the auto build on docker hub; but migrating to this would/will probably move the build to Github Actions anyway.